src/util/router.js

```diff
@@ -79,6 +79,49 @@ export const forceReplace = ({ router, unsavedChanges }, location) => {
   return router.replace(location);
 };
 
+// Returns a scrollBehavior function to pass to createRouter().
+export const createScrollBehavior = (animation = 'smooth') => {
+  // Function to cancel a previous attempt to scroll
+  let cancel;
+  const scrollTo = (scrollId) => {
+    const el = document.querySelector(`[data-scroll-id="${scrollId}"]`);
+    return el != null ? { el, top: 10, behavior: animation } : null;
+  };
+  return (to, from, savedPosition) => {
+    if (cancel != null) cancel();
+    if (to.hash === '') return to.path === from.path ? savedPosition : null;
```

Comment:
Shouldn't we return `savedPosition` even when `to.path` and `from.path` are not equal?

Did you try browser back/forward buttons, are they working correctly?

Reply:
> Did you try browser back/forward buttons, are they working correctly?

Yes, they're working! It's fun to see. `savedPosition` also seems to be working.

> Shouldn't we return `savedPosition` even when `to.path` and `from.path` are not equal?

My thinking was that if you navigate to a different page, it could be that `savedPosition` no longer exists on the page, if data has been cleared and needs to be reloaded. Or even worse, it could be that `savedPosition` exists, but it's a different element and basically just a random place on the page. It could be a different element while data is loading, or even after, depending on how the data has changed. Maybe this isn't super likely though and is outweighed by the convenience of being able to return to `savedPosition` when it does still exist. What do you think?

Reply:
If I understood following correctly then browser should remember scroll position on back/forward buttons:

> The third argument, savedPosition, is only available if this is a popstate navigation (triggered by the browser's back/forward buttons).

I checked out this branch locally, and here is the result. Back button keeps the scroll position from the current page (it should be scroll position of previous page - somewhere in the middle) and forward button is not remembering my scroll position (it should be at the bottom).

https://github.com/getodk/central-frontend/assets/447837/af690a17-8e02-4f03-a475-478345469523

Reply:
Hmm interesting! Maybe I only tried the back button with `savedPosition`. I'd have to look further into it to figure out what's driving the difference. I'd be happy to file an issue about that, but it's probably not something I'd prioritize until we finish the rest of the entity stories.